src/GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -50,17 +59,17 @@ static BranchConfig InheritBranchConfiguration(GitVersionContext context, Branch
                 {
                     excludedInheritBranches = repository.Branches.Where(b =>
                     {
-                        var branchConfig = config.GetConfigForBranch(b.FriendlyName);
+                        var branchConfig = config.GetConfigForBranch(b.NameWithoutRemote());
```

Comment:
This I think is the real fix. This will now exclude local and remote branches which do not have config, or the config is Inherit.

Previously we only excluded branches which had config and it was Inherit, but our infinite loops were caused by branches without config..